### DISCUSSION TITLE
Examples of tldraw extended in the wild?

### DISCUSSION THREAD

**Discussion by hdoro** [tldraw/tldraw-v1#557] [Q&A] [answered]
tldraw is apparently being used by [LegendKeeper](https://www.legendkeeper.com/) and other real-world, production-ready products as the core infrastructure for infinite canvases. Although I understand this is a possibility, I don't yet see how to pull it off myself...

Is there any documentation or open-source examples I could take a look at to start tinkering? Specifically, my biggest burning question is: **should I build on top of @tldraw/tldraw or fork the repo & customize internals**?

**Comment by deadcoder0904** [comment]
yes, would love to know more as wel. that was my initial thought when i checked out the repo. i thought it was just a whiteboarding app but looks like you can do a lot of stuff. there are some examples in the `/examples` folder but would love to know more...

**Comment by steveruizok** [comment, marked as answer]
A good recent example where tldraw is being used as an engine for the app is [okso.app](https://okso.app/). If you'd like to see a more lightweight integration, check out [PopStage](https://popstage.com).

> should I build on top of @tldraw/tldraw or fork the repo & customize internals?

It depends what you want to customize! If you just want the tldraw experience in your app, then you can use the [tldraw package](https://www.npmjs.com/package/@tldraw/tldraw) on NPM. Deeper customizations require forking the project. 

We've been also working on a rewrite of tldraw (almost done!) that allows for customization without needing to fork the project. That should be announced in September and released later this year.

**Reply by deadcoder0904** [reply]
[okso.app](https://okso.app) looks cool! would love to see open source project so we can see the code as i can't find docs.

so this is pure canvas, right? same as konva? i can't use something like [css-doodle](https://css-doodle.com/) right?

**Reply by deadcoder0904** [reply]
as you said in discord which i only saw now:

> konva is html canvas to render, tldraw uses html and css (via React) to render

which means i can use css-doodle bcz it works with html & css but not canvas. correct me if i'm wrong :)

**Comment by huyz** [comment]
The upcoming [whiteboard](https://twitter.com/pengx17/status/1552172906146398208) in LogSeq seems to be another example: https://twitter.com/pengx17/status/1552172912412590080
